src/Akka.Cluster.Hosting.Tests/ClusterSingletonWithDiSpecs.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Hosting;
+using FluentAssertions;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Akka.Cluster.Hosting.Tests;
+
+public class ClusterSingletonWithDiSpecs : Akka.Hosting.TestKit.TestKit
+{
+    #region Actor and DI impls
+    
+    
+    public interface IMyThing
+    {
+        string ThingId { get; }
+    }
+
+    public sealed class ThingImpl : IMyThing
+    {
+        public ThingImpl(string thingId)
+        {
+            ThingId = thingId;
+        }
+
+        public string ThingId { get; }
+    }
+    
+    private class MySingletonDiActor : ReceiveActor
+    {
+        private readonly IMyThing _thing;
+        
+        public MySingletonDiActor(IMyThing thing)
+        {
+            _thing = thing;
+            ReceiveAny(_ => Sender.Tell(_thing.ThingId));
+        }
+    }
+    
+    #endregion
+
+    private readonly TaskCompletionSource _tcs = new(TimeSpan.FromSeconds(3));
+    
+    public ClusterSingletonWithDiSpecs(ITestOutputHelper output) : base(output: output)
+    {
+    }
+
+    protected override void ConfigureServices(HostBuilderContext context, IServiceCollection services)
+    {
+        services.AddSingleton<IMyThing>(new ThingImpl("foo1"));
+        base.ConfigureServices(context, services);
+    }
+
+    protected override void ConfigureAkka(AkkaConfigurationBuilder builder, IServiceProvider provider)
+    {
+        builder.ConfigureHost(configurationBuilder =>
+        {
+            configurationBuilder.WithSingleton<MySingletonDiActor>("my-singleton",
+                (_, _, dependencyResolver) => dependencyResolver.Props<MySingletonDiActor>());
+        },  new ClusterOptions(){ Roles = new[] { "my-host" }}, _tcs, Output!);
+    }
+
+    [Fact]
+    public async Task Should_launch_ClusterSingletonAndProxy_with_DI_delegate()
```

Comment:
Validate that we can start singletons with DI and resolve dependencies correctly afterwards